### PR TITLE
azurite: init at 3.31.0

### PR DESCRIPTION
**Pull request by danielalvsaaker**
## Description of changes
[Azurite](https://github.com/Azure/Azurite) is a lightweight server clone of Azure Storage that simulates most of the commands supported by it with minimal dependencies.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by h7x4 — COMMENTED** [review]
Hello, welcome to nixpkgs :tada:

File: pkgs/by-name/az/azurite.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    owner = "Azure";
+    repo = "Azurite";
+    rev = "v${version}";
+    hash = "sha256-WT1eprN6SBnsfITCopybEHCuxrHvKEhdmVs7xL3cUi0=";
+  };
+
+  npmDepsHash = "sha256-+ptjsz2MDIB/aqu4UxkBLCcehtamFdmswNUsHs23LuE=";
+
+  nativeBuildInputs = [ pkg-config python3 ];
+  buildInputs = [ libsecret ];
```
[the remark this answers is not in the document]
Comment (danielalvsaaker) [reply pkgs/by-name/az/azurite.nix]: It is required by the transitive npm dependency `keytar`. The Dockerfile in the Azurite repo builds from `node:20-alpine3.17`, which seems to transitively include `libsecret` via `gpg`.

[the remark this answers is not in the document]
Comment (danielalvsaaker) [reply pkgs/by-name/az/azurite.nix]: But the reason for including it here is simply that the derivation fails to build without.

[the remark this answers is not in the document]
Comment (h7x4) [reply pkgs/by-name/az/azurite.nix]: Cool, thanks!

**Review by h7x4 — APPROVED** [review]
I have verified your github id, verified the license, nix expression looks clean.

Built and verified that it runs on x86_64-linux nixos.

Ping me for merge once ofborg is finished

**Comment by h7x4** [comment]
Seems like ofborg had a bad time building it on darwin. Would you like to mark it as broken for the time being, and continue with merge, or would you rather have a go at fixing it?

**Comment by danielalvsaaker** [comment]
I'd like to fix it!

**Comment by danielalvsaaker** [comment]
ping @h7x4, ofborg is finally satisfied

**Comment by h7x4** [comment]
Neat! Thank you for contributing. Happy to see a fellow Norwegian around here :)

Feel free to add a backport label if you'd like to backport it to 24.05

**Pull request merged**